### PR TITLE
build_helper: Optionally allow building without jemalloc

### PR DESCRIPTION
**Pull request by akshayknarayan**
As a workaround for https://github.com/facebook/folly/issues/976, modify build_helper.sh to allow disabling jemalloc use by passing the appropriate argument to the cmake invocation on folly.

### CODE REVIEW

File: build_helper.sh [hunk 1]
```diff
@@ -46,6 +46,12 @@ if [ -z "${BUILD_DIR-}" ] ; then
   mkdir -p $BUILD_DIR
 fi
 
+if [[ ! -z $MVFST_FOLLY_USE_JEMALLOC ]]; then
```
Comment (udippant) [inline build_helper.sh]: You can add a flag option on the line # 24 as well (and update the usage)

**Comment by akshayknarayan** [comment]
Looks like the travis build is running into a time limit: https://travis-ci.com/facebookincubator/mvfst/builds/111963309#L9510

**Comment by udippant** [comment]
@akshayknarayan you'll need to sign the CLA before I can import your commit.

**Comment by akshayknarayan** [comment]
@udippant is there something further I need to do? I have emailed cla@fb.com but haven't gotten a response.

**Comment by JoelMarcey** [comment]
@akshayknarayan - your CLA status should be resolved now.

File: build_helper.sh (hunk 1 above)
Comment (udippant) [reply build_helper.sh]: nit: Can you use `-n` instead of `! -z`? 
Looks good overall.

**Pull request closed without merging**